src/libs/SubscriptionUtils.ts

```diff
@@ -0,0 +1,133 @@
+    if (!lastDayFreeTrial) {
+        return 0;
+    }
+
+    const currentDate = new Date();
+    const difference = differenceInCalendarDays(parseDate(lastDayFreeTrial, CONST.DATE.FNS_DATE_TIME_FORMAT_STRING, currentDate), currentDate);
+    return difference < 0 ? 0 : difference;
+}
+
+/**
+ * Whether the workspace's owner is on its free trial period.
+ */
+function isUserOnFreeTrial(): boolean {
+    if (!firstDayFreeTrial || !lastDayFreeTrial) {
+        return true;
+    }
+
+    const currentDate = new Date();
+    const firstDayFreeTrialDate = parseDate(firstDayFreeTrial, CONST.DATE.FNS_DATE_TIME_FORMAT_STRING, currentDate);
+    const lastDayFreeTrialDate = parseDate(lastDayFreeTrial, CONST.DATE.FNS_DATE_TIME_FORMAT_STRING, currentDate);
+
+    return isAfter(currentDate, firstDayFreeTrialDate) && isBefore(currentDate, lastDayFreeTrialDate);
+}
+
+/**
+ * Whether the workspace owner's free trial period has ended.
+ */
+function hasUserFreeTrialEnded(): boolean {
+    if (!lastDayFreeTrial) {
+        return false;
+    }
+
+    const currentDate = new Date();
+    const lastDayFreeTrialDate = parseDate(lastDayFreeTrial, CONST.DATE.FNS_DATE_TIME_FORMAT_STRING, currentDate);
+
+    return isAfter(currentDate, lastDayFreeTrialDate);
+}
+
+/**
+ * Whether the user has a payment card added to its account.
+ */
+function doesUserHavePaymentCardAdded(): boolean {
+    return userBillingFundID !== undefined;
+}
+
+/**
+ * Whether the user's billable actions should be restricted.
+ */
+function shouldRestrictUserBillableActions(policyID: string): boolean {
+    const currentDate = new Date();
+
+    // This logic will be executed if the user is a workspace's non-owner (normal user or admin).
+    // We should restrict the workspace's non-owner actions if it's member of a workspace where the owner is
+    // past due and is past its grace period end.
+    for (const userBillingGraceEndPeriodEntry of Object.entries(userBillingGraceEndPeriodCollection ?? {})) {
+        const [entryKey, userBillingGracePeriodEnd] = userBillingGraceEndPeriodEntry;
+
+        if (userBillingGracePeriodEnd && isAfter(currentDate, fromUnixTime(userBillingGracePeriodEnd.value))) {
+            // Extracts the owner account ID from the collection member key.
+            const ownerAccountID = entryKey.slice(ONYXKEYS.COLLECTION.SHARED_NVP_PRIVATE_USER_BILLING_GRACE_PERIOD_END.length);
+
+            const ownerPolicy = allPolicies?.[`${ONYXKEYS.COLLECTION.POLICY}${policyID}`];
+            if (String(ownerPolicy?.ownerAccountID ?? -1) === ownerAccountID) {
+                return true;
+            }
+        }
+    }
+
+    // If it reached here it means that the user is actually the workspace's owner.
+    // We should restrict the workspace's owner actions if it's past its grace period end date and it's owing some amount.
+    if (ownerBillingGraceEndPeriod && amountOwed !== undefined && isAfter(currentDate, fromUnixTime(ownerBillingGraceEndPeriod))) {
```

Comment:
Can `amountOwed` have a value of `0`? In that case, it will be considered that the workspace owner owes some amount (as we only have `undefined` check) thereby restricting user billable actions. This does not seem correct.

Reply:
Makes sense, I've adjusted but could you confirm @chiragsalian ?